modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/transcripts/TranscriptViewModel.kt

```diff
@@ -152,15 +147,42 @@ class TranscriptViewModel @Inject constructor(
         }
     }
 
-    private suspend fun buildDisplayInfo(cuesWithTimingSubtitle: List<CuesWithTiming>) = withContext(ioDispatcher) {
+    private suspend fun buildDisplayInfo(
+        cuesWithTimingSubtitle: List<CuesWithTiming>,
+        transcriptFormat: TranscriptFormat?,
+    ) = withContext(ioDispatcher) {
+        var previousSpeaker = ""
+        val speakerIndices = mutableListOf<Int>()
         val formattedText = buildString {
             cuesWithTimingSubtitle
                 .flatMap { it.cues }
-                .forEach { cue -> append(cue.text) }
+                .forEach { cue ->
+                    // Extract speaker
+                    val markupText = if (transcriptFormat == TranscriptFormat.VTT) cue.markup else cue.text
+                    markupText?.let {
+                        TranscriptRegexFilters.extractSpeaker(it.toString(), transcriptFormat)?.let { speaker ->
+                            if (previousSpeaker != speaker) {
+                                append("\n\n$speaker\n\n")
```

Comment:
```suggestion
                                append("\n\n")
                                append(speaker)
                                append("\n\n")
```
Feel free to ignore this but should the appends be separate as the StringBuilder is doing them.

Reply:
Suggestion applied in 947faac.